01/app.js

```diff
@@ -11,3 +13,27 @@ form.addEventListener("submit", function (e) {
   newPerson.innerText = name + " " + surname;
   list.appendChild(newPerson);
 });
+*/
+
+// 2)
+
+document.addEventListener("DOMContentLoaded", init);
+
+function init() {
+  const form = document.querySelector("form");
+}
+
+const form = document.querySelector("form");
```

Comment:
Ten zapis jest nieprawidłowy ponieważ szukasz elementu przed załadowaniem DOM.
Cały ten kod możesz wkleić wew. init i będzie działać lub... stworzyć osobną funkcję np.:

```
function init() {
  const form = document.querySelector("form");
  form.addEventListener("submit", handleSubmit);
}

function handleSubmit(e) {
  e.preventDefault();

  const name = e.target.elements.firstName.value;
  const surname = e.target.elements.lastName.value;

  const newPerson = document.createElement("li");
  newPerson.innerText = name + " " + surname;
  const list = document.querySelector(".users-list");
  list.appendChild(newPerson);
}
```

Można też to zrobić w ten sposób (wyszukać element wcześniej i przekazać go przez parametr), ale to już bardziej zaawansowane rozwiązanie:
```
function init() {
  const form = document.querySelector("form");
  const list = document.querySelector(".users-list"); // to przekazuje do wnętrza funkcji
  // dzięki .bind() przekazuje do parametru odpowiednie dane
  // więcej tutaj: https://javascript.info/bind
  form.addEventListener("submit", handleSubmit.bind(null, list));
}

function handleSubmit(personsList, e) {
  e.preventDefault();

  const name = e.target.elements.firstName.value;
  const surname = e.target.elements.lastName.value;

  const newPerson = document.createElement("li");
  newPerson.innerText = name + " " + surname;
  personsList.appendChild(newPerson);
}
```

Reply:
Dziekuje. O bind juz czytalam, ale jeszcze nie potrafie sie nim poslugiwac zbyt dobrze. 
Pisalam wczesniej kod dokladnie taki jak w Twoim pierwszym przykladzie (ta czesc gdzie napisalam "a jesli wszystko umieszcze w srodku tej event listenera dla DOMContentLoad to nic nie dziala? :)" Teraz widze ze wszystko dziala jesli jest tak jak u Ciebie, wszystko dziala jesli wstawie funkcje anonimowa pod handleSubmit a nie dziala tak jak pisalam wczesniej czyli: 
`const init = function () {
  const form = document.querySelector("form");
  form.addEventListener("submit", handleSubmit);
};`
Czyli zapisywalam funkcje w zmiennej. Czy jest jakas przewaga w deklarowaniu funkcji tak jak Ty to robisz albo czy to ze ja zawsze zapisuje je w funkcjach jest nieprawidlowe? Czy do roznych rzeczy powinno sie tego roznie uzywac? I dlaczego jak funkcja zostala zapisana w zmiennej to caly kod nie zadzialal a zaklarowanie jej w ten inny sposob rozwiazuje problem?